### PR TITLE
Allow for creation of new data with a form field input

### PR DESCRIPTION
**Pull request by howhelan**
Add two new query params to the unsecure and secure GET endpoints:
- `create`: Boolean flag indicating whether a GET call is intended to create new data.  Similar to the `edit` param, however no previously existing data with a matching path is required.
- `data_type`:  A DataType Enum, which indicates the type of the new data being created. Required when `create=true`.

Also adding the query param CSS to the `unsecure.handlebars` to allow for styling of the iframe itself (mainly used to remove the iframe border, not sure if there are other use cases or if we can just hard code a no-border iframe.)

### CODE REVIEW

**Comment by capt-haddock** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **howhelan**
You can assign the PR to them by writing `/assign @howhelan` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://jenkins-x.io/v3/develop/reference/chatops/?repo=pauwels-labs%2Fredact-client).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/pauwels-labs/redact-client/blob/main/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by capt-haddock** [comment]
@howhelan: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
build-alpine | 7fc3204d1a6c06d8f3ab4fef30611e11195500a2 | [link](http://dashboard.dev.pauwelslabs.com/namespaces/jx/pipelineruns/build-alpine-jkpjh) | `/test build-alpine`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by capt-haddock** [comment]
@howhelan: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
build-alpine | 7fc3204d1a6c06d8f3ab4fef30611e11195500a2 | [link](http://dashboard.dev.pauwelslabs.com/namespaces/jx/pipelineruns/build-alpine-jkpjh) | `/test build-alpine`
build-prod | 7fc3204d1a6c06d8f3ab4fef30611e11195500a2 | [link](http://dashboard.dev.pauwelslabs.com/namespaces/jx/pipelineruns/build-prod-dtm8m) | `/test build-prod`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by capt-haddock** [comment]
@howhelan: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
build-alpine | 7fc3204d1a6c06d8f3ab4fef30611e11195500a2 | [link](http://dashboard.dev.pauwelslabs.com/namespaces/jx/pipelineruns/build-alpine-jkpjh) | `/test build-alpine`
build-prod | 7fc3204d1a6c06d8f3ab4fef30611e11195500a2 | [link](http://dashboard.dev.pauwelslabs.com/namespaces/jx/pipelineruns/build-prod-dtm8m) | `/test build-prod`
unit-test | 7fc3204d1a6c06d8f3ab4fef30611e11195500a2 | [link](http://dashboard.dev.pauwelslabs.com/namespaces/jx/pipelineruns/unit-test-sgvk7) | `/test unit-test`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by capt-haddock** [comment]
@howhelan: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
build-prod | 7fc3204d1a6c06d8f3ab4fef30611e11195500a2 | [link](http://dashboard.dev.pauwelslabs.com/namespaces/jx/pipelineruns/build-prod-dtm8m) | `/test build-prod`
unit-test | 7fc3204d1a6c06d8f3ab4fef30611e11195500a2 | [link](http://dashboard.dev.pauwelslabs.com/namespaces/jx/pipelineruns/unit-test-sgvk7) | `/test unit-test`
build-alpine | 78d5cb66e1889de43d00eecd5b7eaa6e8e22b52a | [link](http://dashboard.dev.pauwelslabs.com/namespaces/jx/pipelineruns/build-alpine-b5gcc) | `/test build-alpine`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by capt-haddock** [comment]
@howhelan: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
unit-test | 7fc3204d1a6c06d8f3ab4fef30611e11195500a2 | [link](http://dashboard.dev.pauwelslabs.com/namespaces/jx/pipelineruns/unit-test-sgvk7) | `/test unit-test`
build-alpine | 78d5cb66e1889de43d00eecd5b7eaa6e8e22b52a | [link](http://dashboard.dev.pauwelslabs.com/namespaces/jx/pipelineruns/build-alpine-b5gcc) | `/test build-alpine`
build-prod | 78d5cb66e1889de43d00eecd5b7eaa6e8e22b52a | [link](http://dashboard.dev.pauwelslabs.com/namespaces/jx/pipelineruns/build-prod-z55j7) | `/test build-prod`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by capt-haddock** [comment]
@howhelan: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
build-prod | 78d5cb66e1889de43d00eecd5b7eaa6e8e22b52a | [link](http://dashboard.dev.pauwelslabs.com/namespaces/jx/pipelineruns/build-prod-z55j7) | `/test build-prod`
build-alpine | dbe7998e4054cf5852db86a2a4751e4ce2b71c72 | [link](http://dashboard.dev.pauwelslabs.com/namespaces/jx/pipelineruns/build-alpine-qwcch) | `/test build-alpine`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by capt-haddock** [comment]
@howhelan: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
build-prod | 78d5cb66e1889de43d00eecd5b7eaa6e8e22b52a | [link](http://dashboard.dev.pauwelslabs.com/namespaces/jx/pipelineruns/build-prod-z55j7) | `/test build-prod`
build-alpine | e1e3ce73518806288c2a9e798d9d15854ba1173a | [link](http://dashboard.dev.pauwelslabs.com/namespaces/jx/pipelineruns/build-alpine-hr48p) | `/test build-alpine`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

File: src/routes/data/get.rs
Comment (ajpauwels) [inline src/routes/data/get.rs:258]: too much cloning?

File: src/routes/data/get.rs
Comment (ajpauwels) [inline src/routes/data/get.rs:171]: clones necessary?

File: src/routes/data/get.rs
Comment (ajpauwels) [inline src/routes/data/get.rs:41]: rename to data_type

**Review by ajpauwels — CHANGES_REQUESTED** [review]

**Comment by capt-haddock** [comment]
@howhelan: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
build-alpine | e1e3ce73518806288c2a9e798d9d15854ba1173a | [link](http://dashboard.dev.pauwelslabs.com/namespaces/jx/pipelineruns/build-alpine-hr48p) | `/test build-alpine`
build-prod | 207db31b2aa803a5b84eca7f00a87abf6762ab5d | [link](http://dashboard.dev.pauwelslabs.com/namespaces/jx/pipelineruns/build-prod-7xw72) | `/test build-prod`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by capt-haddock** [comment]
@howhelan: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
build-prod | 207db31b2aa803a5b84eca7f00a87abf6762ab5d | [link](http://dashboard.dev.pauwelslabs.com/namespaces/jx/pipelineruns/build-prod-7xw72) | `/test build-prod`
build-alpine | 207db31b2aa803a5b84eca7f00a87abf6762ab5d | [link](http://dashboard.dev.pauwelslabs.com/namespaces/jx/pipelineruns/build-alpine-dqvgb) | `/test build-alpine`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by capt-haddock** [comment]
@howhelan: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
build-alpine | 207db31b2aa803a5b84eca7f00a87abf6762ab5d | [link](http://dashboard.dev.pauwelslabs.com/namespaces/jx/pipelineruns/build-alpine-dqvgb) | `/test build-alpine`
build-prod | 85988ff70823c11d6c6d6467b427e9cc13dd51e4 | [link](http://dashboard.dev.pauwelslabs.com/namespaces/jx/pipelineruns/build-prod-5cpq9) | `/test build-prod`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by capt-haddock** [comment]
@howhelan: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
build-prod | 85988ff70823c11d6c6d6467b427e9cc13dd51e4 | [link](http://dashboard.dev.pauwelslabs.com/namespaces/jx/pipelineruns/build-prod-5cpq9) | `/test build-prod`
build-alpine | 85988ff70823c11d6c6d6467b427e9cc13dd51e4 | [link](http://dashboard.dev.pauwelslabs.com/namespaces/jx/pipelineruns/build-alpine-88lzn) | `/test build-alpine`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by capt-haddock** [comment]
@howhelan: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
build-alpine | 85988ff70823c11d6c6d6467b427e9cc13dd51e4 | [link](http://dashboard.dev.pauwelslabs.com/namespaces/jx/pipelineruns/build-alpine-88lzn) | `/test build-alpine`
build-prod | 784af5cfbdb85c0a71c82f5d943efa7d7c534b8c | [link](http://dashboard.dev.pauwelslabs.com/namespaces/jx/pipelineruns/build-prod-7n4h5) | `/test build-prod`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by capt-haddock** [comment]
@howhelan: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
build-prod | 784af5cfbdb85c0a71c82f5d943efa7d7c534b8c | [link](http://dashboard.dev.pauwelslabs.com/namespaces/jx/pipelineruns/build-prod-7n4h5) | `/test build-prod`
build-alpine | 784af5cfbdb85c0a71c82f5d943efa7d7c534b8c | [link](http://dashboard.dev.pauwelslabs.com/namespaces/jx/pipelineruns/build-alpine-kmnk6) | `/test build-alpine`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Review by ajpauwels — APPROVED** [review]

**Comment by capt-haddock** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *[ajpauwels](https://github.com/pauwels-labs/redact-client/pull/25#pullrequestreview-688844363 "Approved")*

The full list of commands accepted by this bot can be found [here](https://jenkins-x.io/v3/develop/reference/chatops/?repo=pauwels-labs%2Fredact-client).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/pauwels-labs/redact-client/blob/main/OWNERS)~~ [ajpauwels]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**
